tests/unit/modules/network/vyos/test_vyos_bgp_address_family.py

```diff
@@ -497,9 +504,12 @@ def test_vyos_bgp_address_family_rendered(self):
         )
 
         rendered_cmds = [
+            "set protocols bgp 65536 address-family ipv4-unicast network 192.1.13.0/24",
```

Comment:
We now have two types of parsers for redistribute and network (as they can have optional attributes)  configuration. The previous ones, e.g. network.backdoor, tended to remove the entire setting if backdoor attribute was omitted. I built more specific parser and remval expressions, so we do not need a second run for the playbook if we omitted the attribute (otherwise, backdoor parser would remove all, and generic will need a second run to create setting without the attribute)

Reply:
Yes, I understand what was done. I just don't believe it's creating the desired effect. There is code throughout many of the modules to make sure we don't issue redundant commands. We should be able to achieve that same effect here instead of regressing to emitting the redundant commands, especially when there are multiple options for the same parent.

Reply:
This one was solved by introducing setval and remval "smart" functions (similarly to previously used *.metric, etc). Unit tests are now OK with the previous set of commands. 
I plan to render a few more test cases to demo stanzas variation, then clean up unused (commented) stuff and finally re-request a review